models/DataObject/ClassDefinition/Helper/ClassResolver.php

```diff
@@ -37,20 +37,16 @@ protected static function resolve($class, callable $validationCallback = null)
                 return self::$cache[$class];
             }
 
-            try {
-                if (strpos($class, '@') === 0) {
-                    $serviceName = substr($class, 1);
-                    $service = \Pimcore::getKernel()->getContainer()->get($serviceName);
-                } else {
-                    $service = new $class;
-                }
+            if (strpos($class, '@') === 0) {
```

Comment:
```suggestion
            if (str_starts_with($class, '@')) {
```

Reply:
Actually, the whole method could be written as:

```php
    protected static function resolve($class, callable $validationCallback = null)
    {
        if (!$class) {
            return null;
        }

        return self::$cache[$class] ??= self::returnValidServiceOrNull(
            str_starts_with($class, '@') ? \Pimcore::getContainer()->get(substr($class, 1)) : new $class, 
            $validationCallback,
        );
    }
```

Reply:
@jdreesen thank you for your feedback! I, too, was thinking about refactoring, but wasn't too sure about it, since the PR is only supposed to remove the `try-catch`.

Should we still use the new implementation? Maybe if we rename the PR to better reflect the change. WDYT?

Reply:
@passioneight I'd go for the new implementation (as it's an improvement anyway) and rename the PR. 
Can you change the PR accordingly? Thanks a lot!

Reply:
Done.
Thanks & credits should go to @jdreesen :-)